ehrql/query_language.py

```diff
@@ -64,6 +64,10 @@ def define_population(self, population_condition):
         ```
         """
         validate_population_definition(population_condition._qm_node)
+        if "population" in self.variables:
+            raise AttributeError(
+                "Population is already defined and cannot be reassigned"
+            )
```

Comment:
I think this should move up one line (above `validate_population_definition`). If someone calls `dataset.define_population` twice, the first time correctly and the second time incorrectly, we want to tell them that they've called `dataset.define_population` twice, not that they called it incorrectly the second time. (And only then, when they've called it correctly, tell them that they've called it twice.)

I also think the error message could be clearer. We know that the population is stored within `dataset.variables["population"]`, but a user doesn't. The message we want to send to them is something like

```
define_population() should be called no more than once
```

Notice that we start with a lower case "d".

Reply:
thanks @iaindillingham, will fix now!